src/common/regions/utils.ts

```diff
@@ -32,6 +46,7 @@ const sortByPopulation = (regions: Region[]): Region[] =>
  */
 export function getAutocompleteRegions(region?: Region): Region[] {
   const { states, metroAreas, counties } = regions;
+  [states, metroAreas, counties].forEach(sortByNameThenPopulation);
```

Comment:
This implementation works, but it mutates the `states`, `metroAreas` and `counties` [in place](https://en.wikipedia.org/wiki/In-place_algorithm) instead or returning a new array of sorted results.

In this particular case, that doesn't cause any problems, but in general, mutating function parameters can make code difficult to debug. Consider creating a copy and sorting the copy. The code might be easier to read if we separate the comparison function too

```ts
function compareRegionsByNameAndPopulation(regionA: Region, regionB: Region) {
  /* ... */
}

function sortByNameAndPopulation = (locations: Region[]) => {
  const sortedLocations = [...locations]; // this creates a copy
  sortedLocations.sort(compareRegionsByNameAndPopulation);
  return sortedLocations;
}

// ...

return concat<Region>(
  sortByNameAndPopulation(states),
  sortByNameAndPopulation(metroAreas),
  sortByNameAndPopulation(counties),
)
```

As I mentioned, this change is not strictly necessary and is more of a coding styling thing, I always try to avoid mutating function arguments when I can!